packages/material-ui/src/Tooltip/Tooltip.js

```diff
@@ -244,18 +245,7 @@ const Tooltip = React.forwardRef(function Tooltip(props, ref) {
     }, [title, childNode, isControlled]);
   }
 
-  const [defaultId, setDefaultId] = React.useState();
-  const id = idProp || defaultId;
-  React.useEffect(() => {
-    if (!open || defaultId) {
-      return;
-    }
-
-    // Fallback to this default id when possible.
-    // Use the random value for client-side rendering only.
-    // We can't use it server-side.
-    setDefaultId(`mui-tooltip-${Math.round(Math.random() * 1e5)}`);
```

Comment:
I dropped the prefix from `useId` because then I'd need to add it to the dependency array, or use a ref to only capture the initial value and I don't think it's really needed.

Reply:
> add it to the dependency array

What's the issue with such an approach? But this sounds fair, do we need to display a prefix in the id? Maybe not.

Reply:
To me it seems a bit weird because it becomes a dynamic value, and so the effect would re-run. But then now that I check for `defaultId` already being set, `setDefaultId` isn't called again, so it's kinda find, but at the same time the user of this hook might ask, "why did my `prefix` not update"?